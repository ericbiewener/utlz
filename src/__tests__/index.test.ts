import path from 'path'
import { defaultSpawnSync, isFile, sleep } from '..'

test('sleep', async () => {
  expect(await sleep()).toBe(undefined)
  expect(await sleep(0, 1)).toBe(1)
})

test('isFile', () => {
  expect(isFile('blah')).toBe(false)
  expect(isFile(path.join(__dirname, 'index.test.ts'))).toBe(true)
})

test('defaultSpawnSync', () => {
  expect(defaultSpawnSync('echo', ['hello'])).toBe('hello\n')
  expect(() => defaultSpawnSync('fakeCmd', ['hello'])).toThrow()
})
