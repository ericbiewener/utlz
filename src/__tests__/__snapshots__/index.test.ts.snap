// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readDirSync 1`] = `
Array [
  Object {
    "isFile": false,
    "itemPath": "/Users/vn0dimt/Repos/Personal/utlz/src/__tests__/readDirSyncTest/dir1",
    "name": "dir1",
  },
  Object {
    "isFile": false,
    "itemPath": "/Users/vn0dimt/Repos/Personal/utlz/src/__tests__/readDirSyncTest/dir2",
    "name": "dir2",
  },
  Object {
    "isFile": true,
    "itemPath": "/Users/vn0dimt/Repos/Personal/utlz/src/__tests__/readDirSyncTest/file1.txt",
    "name": "file1.txt",
  },
  Object {
    "isFile": true,
    "itemPath": "/Users/vn0dimt/Repos/Personal/utlz/src/__tests__/readDirSyncTest/file2.txt",
    "name": "file2.txt",
  },
]
`;
